opm/input/eclipse/Schedule/Action/ASTNode.cpp

```diff
@@ -114,17 +163,60 @@ bool Opm::Action::ASTNode::empty() const
     return this->size() == 0;
 }
 
-void Opm::Action::ASTNode::add_child(const ASTNode& child)
+// ===========================================================================
+// Private member functions
+// ===========================================================================
+
+Opm::Action::Result
+Opm::Action::ASTNode::evalLogicalOperation(const Context& context) const
 {
-    this->children.push_back(child);
+    auto result = Result { this->type == TokenType::op_and };
+
+    for (const auto& child : this->children) {
+        if (this->type == TokenType::op_or) {
+            result |= child.eval(context);
+        }
+        else {
+            result &= child.eval(context);
+        }
+    }
+
+    return result;
+}
+
+Opm::Action::Result
+Opm::Action::ASTNode::evalComparison(const Context& context) const
+{
+    auto v2 = Value {};
+
+    // Special casing of MONTH comparisons where in addition to symbolic
+    // month names we can compare with numeric month indices.  When
+    // conducting such comparisons, the numeric value should be rounded
+    // before comparison.  This means that for example
+    //
+    //   MNTH = 4.3
+    //
+    // should evaluate to true for the month of April (month index = 4).
+    if (this->children.front().func_type == FuncType::time_month) {
+        const auto& rhs = this->children[1];
+
+        v2 = (rhs.type == TokenType::number)
+            ? Value { std::round(rhs.number) }
```

Comment:
round or floor? if round is appropriate i'd suggest to use an example that makes it clearer.

Reply:
> round or floor?

We've used `round()` since this logic was introduced in commit fcad042bd (PR #2699), and I do think `round()` is appropriate here because the index is supposed to compare to the "nearest integer" value of the right-hand side.  As an example, `MNTH = 4.8` should return true for the month of May.  I'll be sure to add an example.